### ISSUE TITLE
In what way is `UnsafeTriangulator` unsafe?

### ISSUE THREAD

**Issue by HalfVoxel** [andywiecko/BurstTriangulator#189]
It looks like it only uses Unity's safe native data containers. But maybe I missed something?

It looks like you are mirroring Unity's naming convention. IIRC, they use the "Unsafe" prefix to indicate that there's less memory safety.

**Comment by andywiecko** [comment]
Hi @HalfVoxel,

This is not related to memory safety issues. The term "unsafe" here is used to emphasize the potential risks associated with using the API located in `andywiecko.BurstTriangulator.LowLevel.Unsafe`. Additionally user has to provide data by himself (including the output data). It is intended for advanced users only. For more details, please refer to the documentation: https://andywiecko.github.io/BurstTriangulator/manual/advanced/unsafe-triangulator.html

There are many configurations and details that can lead to undesirable results. Let's consider the example of planting holes with refinement:

```csharp
            var t = new UnsafeTriangulator<T>();
            var inputWithHoles = new LowLevel.Unsafe.InputData<T>()
            {
                Positions = LakeSuperior.Points.DynamicCast<T>().AsNativeArray(),
                ConstraintEdges = LakeSuperior.Constraints.AsNativeArray(),
                HoleSeeds = LakeSuperior.Holes.DynamicCast<T>().AsNativeArray(),
            };
            var inputWithoutHoles = inputWithHoles;
            inputWithoutHoles.HoleSeeds = default;
            var args = Args.Default(validateInput: false, refineMesh: true, restoreBoundary: true);
            using var triangles = new NativeList<int>(Allocator.Persistent);
            using var outputPositions = new NativeList<T>(Allocator.Persistent);
            using var halfedges = new NativeList<int>(Allocator.Persistent);
            using var constrainedHalfedges = new NativeList<bool>(Allocator.Persistent);
            var output = new LowLevel.Unsafe.OutputData<T> { Triangles = triangles, Halfedges = halfedges, ConstrainedHalfedges = constrainedHalfedges, Positions = outputPositions };
            t.Triangulate(inputWithoutHoles, output, args.With(refineMesh: false), Allocator.Persistent);
            t.PlantHoleSeeds(inputWithHoles, output, args, Allocator.Persistent);
            t.RefineMesh(output, Allocator.Persistent);
```

In the example above, there are many aspects that the user must configure correctly. Additionally "permutation" of the method calls may not be supported. 

I hope this clarifies why I decided to introduce the convention of `LowLevel.Unsafe`.

Best,
Andrzej

**Comment by HalfVoxel** [comment]
Thanks. Good to know. I expected it to be related to C#'s `unsafe` block, like the Unity's `UnsafeX` containers are.

**Comment by andywiecko** [comment]
I plan to add additional documentation and xml summaries for those. I will mention this to resolve any potential confusion.

**Comment by andywiecko** [comment]
Resolved with [v3.2.0](https://github.com/andywiecko/BurstTriangulator/releases/tag/v3.2.0)
